src/sortStudents.ts

```diff
@@ -1,16 +1,48 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades'
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function calculateAverage(grades: number[]): number {
+  const sum = grades.reduce((prev, item) => prev + item, 0);
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return sum / grades.length;
+}
+
+export function sortStudents(students: Student[],
+  sortBy: SortType, order: SortOrder): Student[] {
+  return [...students].sort((a: Student, b: Student) => {
+    switch (sortBy) {
+      case SortType.Age:
+      case SortType.Married:
+        return order === 'asc'
+          ? +a[sortBy] - +b[sortBy]
+          : +b[sortBy] - +a[sortBy];
+      case SortType.Name:
+      case SortType.Surname:
+        return order === 'asc'
+          ? a[sortBy].localeCompare(b[sortBy])
+          : b[sortBy].localeCompare(a[sortBy]);
+      case SortType.AverageGrade:
+        return order === 'asc'
+          ? calculateAverage(a[sortBy]) - calculateAverage(b[sortBy])
+          : calculateAverage(b[sortBy]) - calculateAverage(a[sortBy]);
+      default:
+        return 0;
```

Comment:
Use default case for error handling.
```suggestion
        throw new Error('Incorrect sort type')
```